be/src/olap/rowset/segment_v2/inverted_index_compound_directory.cpp

```diff
@@ -300,17 +300,9 @@ lucene::store::IndexInput* DorisCompoundDirectory::FSIndexInput::clone() const {
 void DorisCompoundDirectory::FSIndexInput::close() {
     BufferedIndexInput::close();
     if (_handle != nullptr) {
-        _handle->_shared_lock.lock();
-
-        //determine if we are about to delete the handle...
-        bool dounlock = (_LUCENE_ATOMIC_INT_GET(_handle->__cl_refcount) > 1);
-
-        //decdelete (deletes if refcount is down to 0
+        std::lock_guard<std::mutex> wlock(_handle->_shared_lock);
```

Comment:
If _CLDECDELETE(_handle) delete _handle on refcount == 0, _handle->_shared_lock mutex will be deleted. It will cause lock_guard unlock a deleted mutex.